# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.h
# Commit: 6c5ad42069d0
# Full Hash: 6c5ad42069d09a8bc8d8306ab4b21c00230091ec
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-07-09 03:44:14
# Description:
#   Bug 1563542 - Correctly align usage of mIsDiscarded and mClosed for BrowsingContext, r=peterv
#   
#   In the bug which introduced mIsDiscarded, the code was changed to not set
#   mClosed during Detach, and only set mIsDiscarded. This was a mistake because a
#   bunch of places are only reading mClosed. Specifically when creating a
# ==============================================================================

diff -r db4c8cbf463e -r 6c5ad42069d0 docshell/base/BrowsingContext.h
--- a/docshell/base/BrowsingContext.h	Fri Jul 05 22:08:13 2019 +0000
+++ b/docshell/base/BrowsingContext.h	Mon Jul 08 17:27:27 2019 +0000
@@ -126,6 +126,11 @@
   // closed.
   bool IsInProcess() const { return mIsInProcess; }
 
+  // Has this BrowsingContext been discarded. A discarded browsing context has
+  // been destroyed, and may not be available on the other side of an IPC
+  // message.
+  bool IsDiscarded() const { return mIsDiscarded; }
+
   // Get the DocShell for this BrowsingContext if it is in-process, or
   // null if it's not.
   nsIDocShell* GetDocShell() { return mDocShell; }